cmd/internal/labeler/labeler.go

```diff
@@ -455,8 +379,113 @@ func createNumaNodeMappingLabel(mapping map[int][]string) string {
 	return strings.Join(parts, "_")
 }
 
-func (l *labeler) printLabels() {
+func (l *labeler) atomicPrintLabelsToFile(labelFile string) error {
```

Comment:
Add TODO to switch from temp dir to dot file after this PR is included to NFD releases, and GPU plugin install dependencies are updated to such version: https://github.com/kubernetes-sigs/node-feature-discovery/pull/1353  ?

Reply:
I'd rather create an issue as this PR has dragged on for so long.

Reply:
I added the todo as I rebased the PR.